src/ucp/wireup/wireup.c

```diff
@@ -473,13 +487,11 @@ ucp_wireup_process_reply(ucp_worker_h worker, const ucp_wireup_msg_t *msg,
     ucp_wireup_remote_connected(ep);
 
     if (ack) {
-        /* Send ACK without any address, we've already sent it as part of the request */
-        ucs_trace("ep %p: sending wireup ack", ep);
-        memset(rsc_tli, -1, sizeof(rsc_tli));
-        status = ucp_wireup_msg_send(ep, UCP_WIREUP_MSG_ACK, 0, rsc_tli);
-        if (status != UCS_OK) {
-            return;
-        }
+        /* Send `UCP_WIREUP_MSG_ACK` from progress function
+         * to avoid calling UCT routines from an async thread */
+        uct_worker_progress_register_safe(worker->uct,
+                                          ucp_wireup_send_msg_ack, ep,
```

Comment:
need to clean this while endpoint is destroyed

Reply:
done